### PR TITLE
Enable mixer to run as nonroot

### PR DESCRIPTION
**Pull request by clyang82**
Signed-off-by: Chun Lin Yang <clyang@cn.ibm.com>

this PR is for #10768

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[x] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: mixer/docker/Dockerfile.mixer [hunk 1]
```diff
@@ -1,22 +1,32 @@
 # BASE_DISTRIBUTION is used to switch between the old base distribution and distroless base images
 ARG BASE_DISTRIBUTION=default
 
+# The following section is used as base image to add users
+# hadolint ignore=DL3006
+FROM istionightly/base_debug as base_debug
+COPY mixs /usr/local/bin/
+# 1001 is chosen as the UID to be added to the istio group as this is the UID that mixs in the docker container runs as.
+RUN addgroup --system --gid 1001 istio && \
```
Comment (howardjohn) [inline mixer/docker/Dockerfile.mixer]: what is the UID of distroless `nonroot`? is it ok if its not the same 1001?

Comment (clyang82) [reply mixer/docker/Dockerfile.mixer]: the UID is 1002 for `nonroot`. I cannot imagine what problems we may have.

**Pull request merged**